Libs/MRML/Logic/vtkMRMLViewLinkLogic.cxx

```diff
@@ -0,0 +1,462 @@
+{
+  if (!node)
+    {
+    return;
+    }
+  if (node->IsA("vtkMRMLViewNode") || node->IsA("vtkMRMLCameraNode"))
+    {
+    vtkEventBroker::GetInstance()->RemoveObservations(
+      node, vtkCommand::ModifiedEvent, this, this->GetMRMLNodesCallbackCommand());
+    }
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLViewLinkLogic::OnMRMLNodeModified(vtkMRMLNode* node)
+{
+  // Update from CameraNode
+  vtkMRMLCameraNode* cameraNode = vtkMRMLCameraNode::SafeDownCast(node);
+  if (cameraNode && cameraNode->GetID() &&
+      this->GetMRMLScene() && !this->GetMRMLScene()->IsBatchProcessing())
+    {
+
+    // if this is not the node that we are interacting with, short circuit
+    if (!cameraNode->GetInteracting() || !cameraNode->GetInteractionFlags())
+      {
+      return;
+      }
+
+    // CameraNode was modified. Need to find the corresponding
+    // ViewNode to check whether operations are linked
+    vtkMRMLViewNode* viewNode = vtkMRMLViewNode::SafeDownCast
+      (this->GetMRMLScene()->GetNodeByID(cameraNode->GetActiveTag()));
+
+    if (viewNode && viewNode->GetLinkedControl())
+      {
+      this->BroadcastCameraNodeEvent(cameraNode);
+      }
+    else
+      {
+      // camera node changed and views are not linked. Do not broadcast.
+      return;
+      }
+    }
+
+  // Update from viewNode
+  vtkMRMLViewNode* viewNode = vtkMRMLViewNode::SafeDownCast(node);
+  if (viewNode && viewNode->GetID() &&
+      this->GetMRMLScene() && !this->GetMRMLScene()->IsBatchProcessing())
+    {
+    // if this is not the node that we are interacting with, short circuit
+    if (!viewNode->GetInteracting()
+        || !viewNode->GetInteractionFlags())
+      {
+      return;
+      }
+
+    if (viewNode && viewNode->GetLinkedControl())
+      {
+      // view node changed and views are linked. Broadcast.
+      this->BroadcastViewNodeEvent(viewNode);
+      }
+    else
+      {
+      // view node changed and views are not linked. Do not broadcast.
+      }
+    }
+}
+
+
+//----------------------------------------------------------------------------
+void vtkMRMLViewLinkLogic::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os, indent);
+  vtkIndent nextIndent;
+  nextIndent = indent.GetNextIndent();
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLViewLinkLogic::BroadcastCameraNodeEvent(vtkMRMLCameraNode *cameraNode)
+{
+  // only broadcast a camera node event if we are not already actively
+  // broadcasting events and we are actively interacting with the node
+  if (!cameraNode || !cameraNode->GetInteracting())
+    {
+    return;
+    }
+
+  vtkMRMLViewNode* viewNode = vtkMRMLViewNode::SafeDownCast
+    (this->GetMRMLScene()->GetNodeByID(cameraNode->GetActiveTag()));
+
+  if (!viewNode)
+    {
+    return;
+    }
+
+  int requiredViewGroup = viewNode->GetViewGroup();
+  vtkMRMLCameraNode* sNode;
```

Comment:
Please always initialize variables everywhere. It will save a lot of trouble (e.g. these may cause a crash in release mode but not in debug).